fanficfare/adapters/adapter_scribblehubcom.py

```diff
@@ -270,8 +270,64 @@ def getChapterText(self, url):
         soup = self.make_soup(self._fetchUrl(url))
 
         div = soup.find('div', {'id' : 'chp_raw'})
-        if div.find('div', {'class' : 'wi_authornotes'}):
-            div.find('div', {'class' : 'wi_authornotes'}).decompose()
+
+        exclude_notes = self.getConfigList('exclude_notes')
+
+        if 'authornotes' in exclude_notes:
+            # Remove author's notes
+            # There can be multiple in one chapter, so loop until none are left
+            while div.find('div', {'class' : 'wi_authornotes'}):
```

Comment:
This would more efficient as:
```
for antag in div.find_all('div', {'class' : 'wi_authornotes'}):
    antag.decompose()
```
As it is, you're searching the soup tree over and over again.

If you are more comfortable with CSS selectors, BeautifulSoup also has `select()` and `select_one()` methods:
```
for antag in div.select('div.wi_authornotes'):
    antag.decompose()
```
(Many of the FFF supported sites predate those methods in BS and I was more comfortable with `find()` and `find_all()` for a long time, which is why you see more `find()`s in FFF.)

Reply:
Ah yeah, that's much cleaner, thanks for pointing it out